# EDA Datasets Youtube

## Datasets Yang Digunakan
</br></br>
Datasets yang digunakan adalah datasets video Youtube yang bersmber dari Kaggle dengan link sebagai berikut : <br>https://www.kaggle.com/datasnaek/youtube-new


Datasets yang digunakan dalam eksplorasi dibatasi menjadi 3 wilayah/region Canada (CA), Great Britain (GB), dan United States (US).


Jenis dari datasets dari ketiga region tersebut adalah sama yaitu : <b>
 - Data video dengan format .csv <br>
 - Data kategori dengan format .json</b>

## Import library dan load datasets
Import library <b>Pandas</b> dan library lain yang diperlukan untuk membaca datasets .csv dan .json. <br/><br/>
Datasets yang diload adalah sebagai berikut : <b>
- CA_category_id.json
- CAvideos.csv
- GB_category_id.json
- GBvideos.csv
- US_category_id.json
- USvideos.csv </b>

Datasets tersebut diubah menjadi dataframe terlebih dahulu dengan library Pandas agar dapat diolah.

In [1]:
#Import library
import pandas as pd
import json as js
from pandas.io.json import json_normalize

#load datasets CA
dfCAvideos = pd.read_csv("datasets/CAvideos.csv")
with open("datasets/CA_category_id.json") as file:    
    jsCAcategory = js.load(file)

#Load datasets GB
dfGBvideos = pd.read_csv("datasets/GBvideos.csv")
with open("datasets/GB_category_id.json") as file:    
    jsGBcategory = js.load(file)
    
#Load datasets US
dfUSvideos = pd.read_csv("datasets/USvideos.csv")
with open("datasets/US_category_id.json") as file:    
    jsUScategory = js.load(file)

# set option agar float dapat ditampilkan seluruhnya
pd.set_option('float_format', '{:f}'.format)

In [2]:
dfCAvideos.dtypes

video_id                  object
trending_date             object
title                     object
channel_title             object
category_id                int64
publish_time              object
tags                      object
views                      int64
likes                      int64
dislikes                   int64
comment_count              int64
thumbnail_link            object
comments_disabled           bool
ratings_disabled            bool
video_error_or_removed      bool
description               object
dtype: object

## Dataset Video
Dataset video adalah dataset video YouTube yang terdiri dari kolom sebagai berikut :
 - <b>video_id</b> (ID Video) </br>
 - <b>trending_date</b> (Tanggal Trending) </br>
 - <b>title</b> (Judul Video) </br>
 - <b>channel_title</b> (Nama Channel)</br>
 - <b>category_id</b> (ID Kategori) </br>
 - <b>publish_time</b> (Tanggal Upload) <br>
 - <b>tags</b> (Tag-tag) </br>
 - <b>views</b> (Jumlah Penonton) </br>
 - <b>likes</b> (Jumlah yang Menyukai Video) </br>
 - <b>dislikes</b> (Jumlah yang Tidak Menyukai Video) </br>
 - <b>comment_count</b> (Jumlah Komentar) </br>
 - <b>thumbnail_link</b> (Link Thumbnail) </br>
 - <b>comments_disabled</b> (Status Kolom Komentar) </br>
 - <b>ratings_disabled</b> (Status Rating) </br>
 - <b>video_error_or_removed</b> (Status Video Error/Dihapus) </br>
 - <b>description</b> (Deskripsi Video)
 
Dataset video dari 3 region adalah sebagai berikut :
- CAvideos.csv (dataset video region Canada)
- GBvideos.csv (dataset video region Great Britain)
- USvideos.csv (dataset video region United States)

Karena ke-3nya berformat .csv maka dapat diubah menjadi dataframe dengan sintaks pd.read_csv seperti yang dilakukan di cell sebelumnya.

### Data Video Region Canada
Berikut adalah data video dari region Canada :

In [3]:
dfCAvideos.head()

video_id trending_date  \
0  n1WpP7iowLc      17.14.11   
1  0dBIkQ4Mz1M      17.14.11   
2  5qpjK5DgCt4      17.14.11   
3  d380meD0W0M      17.14.11   
4  2Vv-BfVoq4g      17.14.11   

                                               title channel_title  \
0         Eminem - Walk On Water (Audio) ft. Beyoncé    EminemVEVO   
1                      PLUSH - Bad Unboxing Fan Mail     iDubbbzTV   
2  Racist Superman | Rudy Mancuso, King Bach & Le...  Rudy Mancuso   
3                           I Dare You: GOING BALD!?      nigahiga   
4        Ed Sheeran - Perfect (Official Music Video)    Ed Sheeran   

   category_id              publish_time  \
0           10  2017-11-10T17:00:03.000Z   
1           23  2017-11-13T17:00:00.000Z   
2           23  2017-11-12T19:05:24.000Z   
3           24  2017-11-12T18:01:41.000Z   
4           10  2017-11-09T11:04:14.000Z   

                                                tags     views    likes  \
0  Eminem|"Walk"|"On"|"Water"|"Aftermath/Shady/In...  17158579   787425   
1  plush|"bad unboxing"|"unboxing"|"fan mail"|"id...   1014651   127794   
2  racist superman|"rudy"|"mancuso"|"king"|"bach"...   3191434   146035   
3  ryan|"higa"|"higatv"|"nigahiga"|"i dare you"|"...   2095828   132239   
4  edsheeran|"ed sheeran"|"acoustic"|"live"|"cove...  33523622  1634130   

   dislikes  comment_count                                  thumbnail_link  \
0     43420         125882  https://i.ytimg.com/vi/n1WpP7iowLc/default.jpg   
1      1688          13030  https://i.ytimg.com/vi/0dBIkQ4Mz1M/default.jpg   
2      5339           8181  https://i.ytimg.com/vi/5qpjK5DgCt4/default.jpg   
3      1989          17518  https://i.ytimg.com/vi/d380meD0W0M/default.jpg   
4     21082          85067  https://i.ytimg.com/vi/2Vv-BfVoq4g/default.jpg   

   comments_disabled  ratings_disabled  video_error_or_removed  \
0              False             False                   False   
1              False             False                   False   
2              False             False                   False   
3              False             False                   False   
4              False             False                   False   

                                         description  
0  Eminem's new track Walk on Water ft. Beyoncé i...  
1  STill got a lot of packages. Probably will las...  
2  WATCH MY PREVIOUS VIDEO ▶ \n\nSUBSCRIBE ► http...  
3  I know it's been a while since we did this sho...  
4  🎧: https://ad.gt/yt-perfect\n💰: https://atlant...

In [4]:
dfCAvideos.shape

(40881, 16)

Berdasarkan pengecekan data video dari region Canada memiliki dimensi <b>40881 baris x 16 kolom</b>.

### Data Video Region Great Britain
Berikut adalah data video dari region Great Britain :

In [5]:
dfGBvideos.head()

video_id trending_date  \
0  Jw1Y-zhQURU      17.14.11   
1  3s1rvMFUweQ      17.14.11   
2  n1WpP7iowLc      17.14.11   
3  PUTEiSjKwJU      17.14.11   
4  rHwDegptbI4      17.14.11   

                                               title  \
0      John Lewis Christmas Ad 2017 - #MozTheMonster   
1          Taylor Swift: …Ready for It? (Live) - SNL   
2         Eminem - Walk On Water (Audio) ft. Beyoncé   
3  Goals from Salford City vs Class of 92 and Fri...   
4  Dashcam captures truck's near miss with child ...   

                channel_title  category_id              publish_time  \
0                  John Lewis           26  2017-11-10T07:38:29.000Z   
1         Saturday Night Live           24  2017-11-12T06:24:44.000Z   
2                  EminemVEVO           10  2017-11-10T17:00:03.000Z   
3  Salford City Football Club           17  2017-11-13T02:30:38.000Z   
4            Cute Girl Videos           25  2017-11-13T01:45:13.000Z   

                                                tags     views   likes  \
0  christmas|"john lewis christmas"|"john lewis"|...   7224515   55681   
1  SNL|"Saturday Night Live"|"SNL Season 43"|"Epi...   1053632   25561   
2  Eminem|"Walk"|"On"|"Water"|"Aftermath/Shady/In...  17158579  787420   
3  Salford City FC|"Salford City"|"Salford"|"Clas...     27833     193   
4                                             [none]      9815      30   

   dislikes  comment_count                                  thumbnail_link  \
0     10247           9479  https://i.ytimg.com/vi/Jw1Y-zhQURU/default.jpg   
1      2294           2757  https://i.ytimg.com/vi/3s1rvMFUweQ/default.jpg   
2     43420         125882  https://i.ytimg.com/vi/n1WpP7iowLc/default.jpg   
3        12             37  https://i.ytimg.com/vi/PUTEiSjKwJU/default.jpg   
4         2             30  https://i.ytimg.com/vi/rHwDegptbI4/default.jpg   

   comments_disabled  ratings_disabled  video_error_or_removed  \
0              False             False                   False   
1              False             False                   False   
2              False             False                   False   
3              False             False                   False   
4              False             False                   False   

                                         description  
0  Click here to continue the story and make your...  
1  Musical guest Taylor Swift performs …Ready for...  
2  Eminem's new track Walk on Water ft. Beyoncé i...  
3  Salford drew 4-4 against the Class of 92 and F...  
4  Dashcam captures truck's near miss with child ...

In [6]:
dfGBvideos.shape

(38916, 16)

Berdasarkan pengecekan data video dari region Great Britain memiliki dimensi <b>38916 baris x 16 kolom</b>.

### Data Video Region Great Britain
Berikut adalah data video dari region Great Britain :

In [7]:
dfUSvideos.head()

video_id trending_date  \
0  2kyS6SvSYSE      17.14.11   
1  1ZAPwfrtAFY      17.14.11   
2  5qpjK5DgCt4      17.14.11   
3  puqaWrEC7tY      17.14.11   
4  d380meD0W0M      17.14.11   

                                               title          channel_title  \
0                 WE WANT TO TALK ABOUT OUR MARRIAGE           CaseyNeistat   
1  The Trump Presidency: Last Week Tonight with J...        LastWeekTonight   
2  Racist Superman | Rudy Mancuso, King Bach & Le...           Rudy Mancuso   
3                   Nickelback Lyrics: Real or Fake?  Good Mythical Morning   
4                           I Dare You: GOING BALD!?               nigahiga   

   category_id              publish_time  \
0           22  2017-11-13T17:13:01.000Z   
1           24  2017-11-13T07:30:00.000Z   
2           23  2017-11-12T19:05:24.000Z   
3           24  2017-11-13T11:00:04.000Z   
4           24  2017-11-12T18:01:41.000Z   

                                                tags    views   likes  \
0                                    SHANtell martin   748374   57527   
1  last week tonight trump presidency|"last week ...  2418783   97185   
2  racist superman|"rudy"|"mancuso"|"king"|"bach"...  3191434  146033   
3  rhett and link|"gmm"|"good mythical morning"|"...   343168   10172   
4  ryan|"higa"|"higatv"|"nigahiga"|"i dare you"|"...  2095731  132235   

   dislikes  comment_count                                  thumbnail_link  \
0      2966          15954  https://i.ytimg.com/vi/2kyS6SvSYSE/default.jpg   
1      6146          12703  https://i.ytimg.com/vi/1ZAPwfrtAFY/default.jpg   
2      5339           8181  https://i.ytimg.com/vi/5qpjK5DgCt4/default.jpg   
3       666           2146  https://i.ytimg.com/vi/puqaWrEC7tY/default.jpg   
4      1989          17518  https://i.ytimg.com/vi/d380meD0W0M/default.jpg   

   comments_disabled  ratings_disabled  video_error_or_removed  \
0              False             False                   False   
1              False             False                   False   
2              False             False                   False   
3              False             False                   False   
4              False             False                   False   

                                         description  
0  SHANTELL'S CHANNEL - https://www.youtube.com/s...  
1  One year after the presidential election, John...  
2  WATCH MY PREVIOUS VIDEO ▶ \n\nSUBSCRIBE ► http...  
3  Today we find out if Link is a Nickelback amat...  
4  I know it's been a while since we did this sho...

In [8]:
dfUSvideos.shape

(40949, 16)

Berdasarkan pengecekan data video dari region United States memiliki dimensi <b>40849 baris x 16 kolom</b>.

<b>------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------</b>

<b>------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------</b>

## Dataset Kategori
Dataset kategori adalah dataset kategori video di Youtube.

Dataset kategori dari 3 region tersebut adalah :
- CA_category_id.json (dataset kategori video region Canada)
- GB_category_id.json (dataset kategori video region Great Britain)
- US_category_id.json (dataset kategori video region United States)

Karena dataset berformat .json maka perlu dilakukan transformasi untuk mengubahnya menjadi dataframe.

Mari melihat isi dari file kategori .json dari region Canada sebagai contoh.

In [9]:
jsCAcategory

{'kind': 'youtube#videoCategoryListResponse',
 'etag': '"ld9biNPKjAjgjV7EZ4EKeEGrhao/1v2mrzYSYG6onNLt2qTj13hkQZk"',
 'items': [{'kind': 'youtube#videoCategory',
   'etag': '"ld9biNPKjAjgjV7EZ4EKeEGrhao/Xy1mB4_yLrHy_BmKmPBggty2mZQ"',
   'id': '1',
   'snippet': {'channelId': 'UCBR8-60-B28hp2BmDPdntcQ',
    'title': 'Film & Animation',
    'assignable': True}},
  {'kind': 'youtube#videoCategory',
   'etag': '"ld9biNPKjAjgjV7EZ4EKeEGrhao/UZ1oLIIz2dxIhO45ZTFR3a3NyTA"',
   'id': '2',
   'snippet': {'channelId': 'UCBR8-60-B28hp2BmDPdntcQ',
    'title': 'Autos & Vehicles',
    'assignable': True}},
  {'kind': 'youtube#videoCategory',
   'etag': '"ld9biNPKjAjgjV7EZ4EKeEGrhao/nqRIq97-xe5XRZTxbknKFVe5Lmg"',
   'id': '10',
   'snippet': {'channelId': 'UCBR8-60-B28hp2BmDPdntcQ',
    'title': 'Music',
    'assignable': True}},
  {'kind': 'youtube#videoCategory',
   'etag': '"ld9biNPKjAjgjV7EZ4EKeEGrhao/HwXKamM1Q20q9BN-oBJavSGkfDI"',
   'id': '15',
   'snippet': {'channelId': 'UCBR8-60-B28hp2BmDPdnt

Berdasarkan pengecekan tersebut, dataset kategori memiliki susunan sebagai berikut :
 - kind<br>
 - etag<br>
 - id<br>
 - items<br>
 - snippet<br>
 - title<br>
 - assignable<br>

Kemudian ditemukan bahwa items adalah suatu list yang berisi kind, etag, id, snippet, title dan assignable.

### Mentransformasi data kategori JSON menjadi dataframe
Untuk keperluan analisis, yang dibutuhkan di sini adalah id kategori dan title / nama kategori. Maka 2 hal tersebut yang akan diekstrak kemudian dimasukkan ke dalam dataframe.

#### Normalisasi list items di data kategori
Karena yang dibutuhkan adalah id dan title di dalam list items, maka items harus dinormalisasi dengan menggunakan sintaks <b>json_normalize()</b>.
Dengan json_normalize, Pandas dapat mengambil objek JSON yang bersarang dan mengubahnya menjadi dataframe.

In [10]:
jsCAcatitem = json_normalize(jsCAcategory['items'])
jsGBcatitem = json_normalize(jsGBcategory['items'])
jsUScatitem = json_normalize(jsUScategory['items'])

Mari ambil contoh items yang telah dinormalisasi dari data kategori region Canada

In [11]:
jsCAcatitem

etag  id  \
0   "ld9biNPKjAjgjV7EZ4EKeEGrhao/Xy1mB4_yLrHy_BmKm...   1   
1   "ld9biNPKjAjgjV7EZ4EKeEGrhao/UZ1oLIIz2dxIhO45Z...   2   
2   "ld9biNPKjAjgjV7EZ4EKeEGrhao/nqRIq97-xe5XRZTxb...  10   
3   "ld9biNPKjAjgjV7EZ4EKeEGrhao/HwXKamM1Q20q9BN-o...  15   
4   "ld9biNPKjAjgjV7EZ4EKeEGrhao/9GQMSRjrZdHeb1OEM...  17   
5   "ld9biNPKjAjgjV7EZ4EKeEGrhao/FJwVpGCVZ1yiJrqZb...  18   
6   "ld9biNPKjAjgjV7EZ4EKeEGrhao/M-3iD9dwK7YJCafRf...  19   
7   "ld9biNPKjAjgjV7EZ4EKeEGrhao/WmA0qYEfjWsAoyJFS...  20   
8   "ld9biNPKjAjgjV7EZ4EKeEGrhao/EapFaGYG7K0StIXVf...  21   
9   "ld9biNPKjAjgjV7EZ4EKeEGrhao/xId8RX7vRN8rqkbYZ...  22   
10  "ld9biNPKjAjgjV7EZ4EKeEGrhao/G9LHzQmx44rX2S5ya...  23   
11  "ld9biNPKjAjgjV7EZ4EKeEGrhao/UVB9oxX2Bvqa_w_y3...  24   
12  "ld9biNPKjAjgjV7EZ4EKeEGrhao/QiLK0ZIrFoORdk_g2...  25   
13  "ld9biNPKjAjgjV7EZ4EKeEGrhao/r6Ck6Z0_L0rG37VJQ...  26   
14  "ld9biNPKjAjgjV7EZ4EKeEGrhao/EoYkczo9I3RCf96Rv...  27   
15  "ld9biNPKjAjgjV7EZ4EKeEGrhao/w5HjcTD82G_XA3xBc...  28   
16  "ld9biNPKjAjgjV7EZ4EKeEGrhao/lL7uWDr_071CHxifj...  30   
17  "ld9biNPKjAjgjV7EZ4EKeEGrhao/WnuVfjO-PyFLO7NTR...  31   
18  "ld9biNPKjAjgjV7EZ4EKeEGrhao/ctpH2hGA_UZ3volJT...  32   
19  "ld9biNPKjAjgjV7EZ4EKeEGrhao/L0kR3-g1BAo5UD1PL...  33   
20  "ld9biNPKjAjgjV7EZ4EKeEGrhao/pUZOAC_s9sfiwar63...  34   
21  "ld9biNPKjAjgjV7EZ4EKeEGrhao/Xb5JLhtyNRN3AQq02...  35   
22  "ld9biNPKjAjgjV7EZ4EKeEGrhao/u8WXzF4HIhtEi805_...  36   
23  "ld9biNPKjAjgjV7EZ4EKeEGrhao/D04PP4Gr7wc4IV_O9...  37   
24  "ld9biNPKjAjgjV7EZ4EKeEGrhao/i5-_AceGXQCEEMWU0...  38   
25  "ld9biNPKjAjgjV7EZ4EKeEGrhao/rtlxd0zOixA9QHdIZ...  39   
26  "ld9biNPKjAjgjV7EZ4EKeEGrhao/N1TrDFLRppxZgBowC...  40   
27  "ld9biNPKjAjgjV7EZ4EKeEGrhao/7UMGi6zRySqXopr_r...  41   
28  "ld9biNPKjAjgjV7EZ4EKeEGrhao/RScXhi324h8usyIet...  42   
29  "ld9biNPKjAjgjV7EZ4EKeEGrhao/0n9MJVCDLpA8q7aiG...  43   
30  "ld9biNPKjAjgjV7EZ4EKeEGrhao/x5NxSf5fz8hn4loSN...  44   

                     kind  snippet.assignable         snippet.channelId  \
0   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
1   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
2   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
3   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
4   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
5   youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
6   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
7   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
8   youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
9   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
10  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
11  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
12  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
13  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
14  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
15  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
16  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
17  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
18  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
19  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
20  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
21  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
22  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
23  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
24  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
25  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
26  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ

#### Memasukkan isi dari items ke dalam dataframe
items dari semua dataset kategori dari ke-3 region dimasukkan menjadi dataframe agar mudah untuk dikelola

In [12]:
dfCAcategory = jsCAcatitem
dfGBcategory = jsGBcatitem
dfUScategory = jsUScatitem

### Mentransformasi Data kategori Menjadi ID dan Nama Kategori Saja
ID dan nama category/snippet title dibutuhkan untuk dimerge dengan dataframe video. Maka harus dilakukan drop pada kolom yang tidak diperlukan pada dataframe.

In [13]:
dfCAcategory1 = dfCAcategory.drop(['etag','kind','snippet.assignable','snippet.channelId'], axis=1)
dfGBcategory1 = dfGBcategory.drop(['etag','kind','snippet.assignable','snippet.channelId'], axis=1)
dfUScategory1 = dfUScategory.drop(['etag','kind','snippet.assignable','snippet.channelId'], axis=1)

In [14]:
dfCAcategory1

id         snippet.title
0    1      Film & Animation
1    2      Autos & Vehicles
2   10                 Music
3   15        Pets & Animals
4   17                Sports
5   18          Short Movies
6   19       Travel & Events
7   20                Gaming
8   21         Videoblogging
9   22        People & Blogs
10  23                Comedy
11  24         Entertainment
12  25       News & Politics
13  26         Howto & Style
14  27             Education
15  28  Science & Technology
16  30                Movies
17  31       Anime/Animation
18  32      Action/Adventure
19  33              Classics
20  34                Comedy
21  35           Documentary
22  36                 Drama
23  37                Family
24  38               Foreign
25  39                Horror
26  40        Sci-Fi/Fantasy
27  41              Thriller
28  42                Shorts
29  43                 Shows
30  44              Trailers

kolom snippet.title direname menjadi title agar tidak membingungkan.

In [15]:
dfCAcategory1.rename(columns={'snippet.title': 'title'}, inplace=True)
dfGBcategory1.rename(columns={'snippet.title': 'title'}, inplace=True)
dfUScategory1.rename(columns={'snippet.title': 'title'}, inplace=True)

### Data Kategori Video Region Canada
Berikut adalah data kategori video region Canada setelah diubah menjadi dataframe :

In [16]:
dfCAcategory1

id                 title
0    1      Film & Animation
1    2      Autos & Vehicles
2   10                 Music
3   15        Pets & Animals
4   17                Sports
5   18          Short Movies
6   19       Travel & Events
7   20                Gaming
8   21         Videoblogging
9   22        People & Blogs
10  23                Comedy
11  24         Entertainment
12  25       News & Politics
13  26         Howto & Style
14  27             Education
15  28  Science & Technology
16  30                Movies
17  31       Anime/Animation
18  32      Action/Adventure
19  33              Classics
20  34                Comedy
21  35           Documentary
22  36                 Drama
23  37                Family
24  38               Foreign
25  39                Horror
26  40        Sci-Fi/Fantasy
27  41              Thriller
28  42                Shorts
29  43                 Shows
30  44              Trailers

In [17]:
dfCAcategory1.shape

(31, 2)

Berdasarkan pengecekan data kategori region Canada memiliki dimensi <b>31 baris x 2 kolom</b>.

### Data Kategori Video Region Great Britain
Berikut adalah data kategori video region Great Britain setelah diubah menjadi dataframe :

In [18]:
dfGBcategory1

id                 title
0    1      Film & Animation
1    2      Autos & Vehicles
2   10                 Music
3   15        Pets & Animals
4   17                Sports
5   18          Short Movies
6   19       Travel & Events
7   20                Gaming
8   21         Videoblogging
9   22        People & Blogs
10  23                Comedy
11  24         Entertainment
12  25       News & Politics
13  26         Howto & Style
14  27             Education
15  28  Science & Technology
16  30                Movies
17  31       Anime/Animation
18  32      Action/Adventure
19  33              Classics
20  34                Comedy
21  35           Documentary
22  36                 Drama
23  37                Family
24  38               Foreign
25  39                Horror
26  40        Sci-Fi/Fantasy
27  41              Thriller
28  42                Shorts
29  43                 Shows
30  44              Trailers

In [19]:
dfGBcategory1.shape

(31, 2)

Berdasarkan pengecekan data kategori region Great Britain memiliki dimensi <b>31 baris x 2 kolom</b>.

### Data Kategori Video Region United States
Berikut adalah data kategori video region Great Britain setelah diubah menjadi dataframe :

In [20]:
dfUScategory1

id                  title
0    1       Film & Animation
1    2       Autos & Vehicles
2   10                  Music
3   15         Pets & Animals
4   17                 Sports
5   18           Short Movies
6   19        Travel & Events
7   20                 Gaming
8   21          Videoblogging
9   22         People & Blogs
10  23                 Comedy
11  24          Entertainment
12  25        News & Politics
13  26          Howto & Style
14  27              Education
15  28   Science & Technology
16  29  Nonprofits & Activism
17  30                 Movies
18  31        Anime/Animation
19  32       Action/Adventure
20  33               Classics
21  34                 Comedy
22  35            Documentary
23  36                  Drama
24  37                 Family
25  38                Foreign
26  39                 Horror
27  40         Sci-Fi/Fantasy
28  41               Thriller
29  42                 Shorts
30  43                  Shows
31  44               Trailers

In [21]:
dfUScategory1.shape

(32, 2)

Berdasarkan pengecekan data kategori region Canada memiliki dimensi <b>31 baris x 2 kolom</b>.

### Persamaan & Perbedaan Data Kategori dari 3 region
Berdasarkan pengecekan dataframe kategori region Canada dan Great Britain adalah identik, sedangkan dataframe kategori dari United States berbeda dari keduanya.

In [22]:
dfCAcategory1.equals(dfGBcategory1)

True

In [23]:
dfGBcategory1.equals(dfUScategory1)

False

In [24]:
dfCek = pd.concat([dfCAcategory1,dfGBcategory1,dfUScategory1]).drop_duplicates(keep=False)
dfCek

id                  title
16  29  Nonprofits & Activism

Perbedaan yang didapatkan dengan melakukan concat adalah dataframe kategori United States memiliki satu title tambahan yaitu "Nonprofits & Activism" pada index ke-16.

Maka dari itu diputuskan bahwa data kategori dari 3 region ini <b>di-merge menjadi 1 dataframe</b> saja dengan pd.merge(x,y) dengan keterangan sebagai berikut :
- x = dataframe kategori US
- y = dataframe kategori CA

### Merge Data Kategori

In [25]:
#merge
dfCategory = pd.merge(dfCAcategory1,dfUScategory1, how="right")

dfCategory

id                  title
0    1       Film & Animation
1    2       Autos & Vehicles
2   10                  Music
3   15         Pets & Animals
4   17                 Sports
5   18           Short Movies
6   19        Travel & Events
7   20                 Gaming
8   21          Videoblogging
9   22         People & Blogs
10  23                 Comedy
11  24          Entertainment
12  25        News & Politics
13  26          Howto & Style
14  27              Education
15  28   Science & Technology
16  30                 Movies
17  31        Anime/Animation
18  32       Action/Adventure
19  33               Classics
20  34                 Comedy
21  35            Documentary
22  36                  Drama
23  37                 Family
24  38                Foreign
25  39                 Horror
26  40         Sci-Fi/Fantasy
27  41               Thriller
28  42                 Shorts
29  43                  Shows
30  44               Trailers
31  29  Nonprofits & Activism

Dataframe 3 region disatukan menjadi dataframe <b>dfCategory</b>.

### Sortir Data Kategori

Setelah dilakukan merge, didapatkan bahwa "id" tidak berurutan. Maka dari itu dilakukan sorting/sortir dengan .sort_values(). Tetapi agar dapat disortir, kolom "id" harus di-state terlebih dahulu bahwa data di kolom tersebut bertipe integer. Setelah, "id" disortir, perlu dilakukan pengindeksan ulang agar index menjadi berurutan kembali.

Berikut adalah <b>Data Kategori Utama</b> :

In [26]:
#Menstate kolom id sebagai data bertipe integer agar dapat disortir
dfCategory['id'] = dfCategory['id'].astype('int')

#Sortir
dfCategory.sort_values("id", ascending=True, inplace=True)

#Pengindexan ulang
dfCategory = dfCategory.reset_index(drop=True)

print("")
print("DATA KATEGORI UTAMA")
dfCategory


DATA KATEGORI UTAMA


id                  title
0    1       Film & Animation
1    2       Autos & Vehicles
2   10                  Music
3   15         Pets & Animals
4   17                 Sports
5   18           Short Movies
6   19        Travel & Events
7   20                 Gaming
8   21          Videoblogging
9   22         People & Blogs
10  23                 Comedy
11  24          Entertainment
12  25        News & Politics
13  26          Howto & Style
14  27              Education
15  28   Science & Technology
16  29  Nonprofits & Activism
17  30                 Movies
18  31        Anime/Animation
19  32       Action/Adventure
20  33               Classics
21  34                 Comedy
22  35            Documentary
23  36                  Drama
24  37                 Family
25  38                Foreign
26  39                 Horror
27  40         Sci-Fi/Fantasy
28  41               Thriller
29  42                 Shorts
30  43                  Shows
31  44               Trailers

In [27]:
dfCategory.shape

(32, 2)

Berdasarkan pengecekan data kategori memiliki dimensi <b>32 baris x 2 kolom</b>

## Statistik Dari Setiap Kategori di Data Video
Kita perlu mengetahui statistik berikut untuk setiap kategori :
#### 1. Data tertinggi, terendah, dan rata-rata viewer
#### 2. Data tertinggi, terendah, dan rata-rata likes
#### 3. Data tertinggi, terendah, dan rata-rata dislikes
#### 5. Data tertinggi, terendah, dan rata-rata jumlah komentar

### Pertama kita perlu mengetahui kategori apa yang ada pada masing-masing region berdasarkan data video

In [28]:
# drop kolom yang tidak perlu
dfCA = dfCAvideos.drop(['trending_date','channel_title','publish_time','tags','thumbnail_link','comments_disabled','ratings_disabled','video_error_or_removed','description'], axis=1)
dfGB = dfGBvideos.drop(['trending_date','channel_title','publish_time','tags','thumbnail_link','comments_disabled','ratings_disabled','video_error_or_removed','description'], axis=1)
dfUS = dfUSvideos.drop(['trending_date','channel_title','publish_time','tags','thumbnail_link','comments_disabled','ratings_disabled','video_error_or_removed','description'], axis=1)

In [29]:
# mengelompokan berdasarkan id kategori
gCAvideo = dfCA.groupby('category_id')
gGBvideo = dfGB.groupby('category_id')
gUSvideo = dfUS.groupby('category_id')

#### Menemukan value unique pada id kategori region Canada
Kategori yang ada pada data video region Canada adalah kategori dengan ID :
 1, 2, 10, 15, 17, 19, 20, 22, 23, 24, 25, 26, 27, 28, 29, 30, 43

In [30]:
# menemukan value yang unik pada id kategori region canada
dfCA.category_id.unique()

a = dfCA.category_id.unique()
a.sort(axis=0)
a

array([ 1,  2, 10, 15, 17, 19, 20, 22, 23, 24, 25, 26, 27, 28, 29, 30, 43],
      dtype=int64)

#### ID Kategori yang tidak ada di region Canada adalah sebagai berikut :

In [31]:
sCat_ca = pd.Series(a)

print (pd.concat([dfCategory['id'],sCat_ca]).drop_duplicates(keep=False).to_string(index=False))

 18
 21
 31
 32
 33
 34
 35
 36
 37
 38
 39
 40
 41
 42
 44


#### Menemukan value unique pada id kategori region Great Britain
Kategori yang ada pada data video region Great Britain adalah kategori dengan ID :
 1, 2, 10, 15, 17, 19, 20, 22, 23, 24, 25, 26, 27, 28, 29, 30, 43

In [32]:
# menemukan value yang unik pada id kategori region great britain
dfGB.category_id.unique()

b = dfGB.category_id.unique()
b.sort(axis=0)
b

array([ 1,  2, 10, 15, 17, 19, 20, 22, 23, 24, 25, 26, 27, 28, 29, 43],
      dtype=int64)

#### Kategori yang tidak ada di region Great Britain adalah sebagai berikut :

In [33]:
sCat_gb = pd.Series(b)

print (pd.concat([dfCategory['id'],sCat_gb]).drop_duplicates(keep=False).to_string(index=False))

 18
 21
 30
 31
 32
 33
 34
 35
 36
 37
 38
 39
 40
 41
 42
 44


#### Menemukan value unique pada id kategori region United States
Kategori yang ada pada data video region United States adalah kategori dengan ID :
 1, 2, 10, 15, 17, 19, 20, 22, 23, 24, 25, 26, 27, 28, 29, 30, 43

In [34]:
# menemukan value yang unik pada id kategori region us
dfUS.category_id.unique()

c = dfUS.category_id.unique()
c.sort(axis=0)
c

array([ 1,  2, 10, 15, 17, 19, 20, 22, 23, 24, 25, 26, 27, 28, 29, 43],
      dtype=int64)

#### Kategori yang tidak ada di region United States adalah sebagai berikut :

In [35]:
sCat_us = pd.Series(c)

print (pd.concat([dfCategory['id'],sCat_us]).drop_duplicates(keep=False).to_string(index=False))

 18
 21
 30
 31
 32
 33
 34
 35
 36
 37
 38
 39
 40
 41
 42
 44


In [36]:
# membuat data list dari array
data_cat_ca = {'id': dfCA.category_id.unique()}
data_cat_gb = {'id': dfGB.category_id.unique()} 
data_cat_us = {'id': dfUS.category_id.unique()} 

# memasukan data kategori ke dataframe
dfCAcat = pd.DataFrame(data_cat_ca)
dfGBcat = pd.DataFrame(data_cat_gb)
dfUScat = pd.DataFrame(data_cat_us)

In [37]:
# melakukan merge data kategori canada dengan dataframe kategori utama
dfCAcat1 = pd.merge(dfCategory,dfCAcat)
print("")
print("DATA KATEGORI DI CANADA")
dfCAcat1


DATA KATEGORI DI CANADA


id                  title
0    1       Film & Animation
1    2       Autos & Vehicles
2   10                  Music
3   15         Pets & Animals
4   17                 Sports
5   19        Travel & Events
6   20                 Gaming
7   22         People & Blogs
8   23                 Comedy
9   24          Entertainment
10  25        News & Politics
11  26          Howto & Style
12  27              Education
13  28   Science & Technology
14  29  Nonprofits & Activism
15  30                 Movies
16  43                  Shows

In [38]:
# melakukan merge data kategori great britain dengan dataframe kategori utama
dfGBcat1 = pd.merge(dfCategory,dfGBcat)
print("")
print("DATA KATEGORI DI GREAT BRITAIN")
dfGBcat1


DATA KATEGORI DI GREAT BRITAIN


id                  title
0    1       Film & Animation
1    2       Autos & Vehicles
2   10                  Music
3   15         Pets & Animals
4   17                 Sports
5   19        Travel & Events
6   20                 Gaming
7   22         People & Blogs
8   23                 Comedy
9   24          Entertainment
10  25        News & Politics
11  26          Howto & Style
12  27              Education
13  28   Science & Technology
14  29  Nonprofits & Activism
15  43                  Shows

In [39]:
# melakukan merge data kategori united states dengan dataframe kategori utama
dfUScat1 = pd.merge(dfCategory,dfUScat)
print("")
print("DATA KATEGORI DI UNITED STATES")
dfUScat1


DATA KATEGORI DI UNITED STATES


id                  title
0    1       Film & Animation
1    2       Autos & Vehicles
2   10                  Music
3   15         Pets & Animals
4   17                 Sports
5   19        Travel & Events
6   20                 Gaming
7   22         People & Blogs
8   23                 Comedy
9   24          Entertainment
10  25        News & Politics
11  26          Howto & Style
12  27              Education
13  28   Science & Technology
14  29  Nonprofits & Activism
15  43                  Shows

In [40]:
dfGBcat1.equals(dfUScat1)

True

In [41]:
dfGBcat1.equals(dfCAcat1)

False

### Ditemukan bahwa kategori video yang ada di data video Great Britain dan United States adalah identik, yaitu :
- 1	(Film & Animation)
- 2	(Autos & Vehicles)
- 10 (Music)
- 15 (Pets & Animals)
- 17 (Sports)
- 19 (Travel & Events)
- 20 (Gaming)
- 22 (People & Blogs)
- 23 (Comedy)
- 24 (Entertainment)
- 25 (News & Politics)
- 26 (Howto & Style)
- 27 (Education)
- 28 (Science & Technology)
- 29 (Nonprofits & Activism)
- 43 (Shows)

### Sedangkan data video Canada memiliki 1 kategori yang berbeda dari Great Britain dan United States, yaitu :
- 30 (Movies)

## Statistik Data Video Region Canada
Di sini kita akan menemukan statistik data video dari region Canada menggunakan list <b>gCAvideo</b>.<br/>
Untuk menemukan rata-rata digunakan fungsi <b>.max()</b> pada data grup id kategori yang diinginkan. Untuk menemukan value tertinggi digunakan <b>.max()</b> dan untuk menemukan value terkecil digunakan <b>.min()</b>

#### 1A. Data Viewer Canada

In [42]:
# memasukkan data ke dalam list
data_view_ca = {
    
    # a adalah hasil sorting array saat menemukan value yang unique dari id kategori data video region Canada
    'id' : a,
    #penonton tertinggi
    'highest_views':[gCAvideo.get_group(items)['views'].max() for items in dfCAcat1['id']],
    
    #penonton terendah
    'lowest_views':[gCAvideo.get_group(items)['views'].min() for items in dfCAcat1['id']],
    
    #penonton rata-rata
    'average_views':[gCAvideo.get_group(1)['views'].mean() for items in dfCAcat1['id']]}

dfViewCA = pd.DataFrame(data_view_ca)

In [43]:
dfViewCA = pd.merge(dfCAcat1,dfViewCA,on="id")

print("")
print("DATA VIEWER SETIAP KATEGORI DI REGION CANADA")
dfViewCA


DATA VIEWER SETIAP KATEGORI DI REGION CANADA


id                  title  highest_views  lowest_views  average_views
0    1       Film & Animation       27696924          2274 1426728.565049
1    2       Autos & Vehicles       11004698          3059 1426728.565049
2   10                  Music       98938809          3201 1426728.565049
3   15         Pets & Animals        5954189          8171 1426728.565049
4   17                 Sports       24117244          1023 1426728.565049
5   19        Travel & Events        4169455          3144 1426728.565049
6   20                 Gaming       14589314          3494 1426728.565049
7   22         People & Blogs       51243149          1187 1426728.565049
8   23                 Comedy       14228882          1295 1426728.565049
9   24          Entertainment      137843120           733 1426728.565049
10  25        News & Politics       11510500          1000 1426728.565049
11  26          Howto & Style       35696481          3564 1426728.565049
12  27              Education       12100921          1141 1426728.565049
13  28   Science & Technology       36778884          3933 1426728.565049
14  29  Nonprofits & Activism       22387656          1898 1426728.565049
15  30                 Movies        5661965        225528 1426728.565049
16  43                  Shows        1386464         42240 1426728.565049

#### 2A. Data Likes Video di Canada

In [44]:
# memasukkan data ke dalam list
data_likes_ca = {
    
    # a adalah hasil sorting array saat menemukan value yang unique dari id kategori data video region Canada
    'id' : a,
    'highest_likes':[gCAvideo.get_group(items)['likes'].max() for items in dfCAcat1['id']],
    'lowest_likes':[gCAvideo.get_group(items)['likes'].min() for items in dfCAcat1['id']],
    'average_likes':[gCAvideo.get_group(items)['likes'].mean() for items in dfCAcat1['id']
                   ]}

dfLikesCA = pd.DataFrame(data_likes_ca)

In [45]:
dfLikesCA = pd.merge(dfCAcat1,dfLikesCA,on="id")

print("")
print("DATA LIKES SETIAP KATEGORI DI REGION CANADA")
dfLikesCA


DATA LIKES SETIAP KATEGORI DI REGION CANADA


id                  title  highest_likes  lowest_likes  average_likes
0    1       Film & Animation         532660             0   37767.962621
1    2       Autos & Vehicles          86125            40   13405.422096
2   10                  Music        5053338             0  151285.856339
3   15         Pets & Animals         193092           215   23931.533875
4   17                 Sports         926660             0   21330.972013
5   19        Travel & Events         135057             4    9568.658163
6   20                 Gaming         340240             0   38487.017857
7   22         People & Blogs        1967910             0   25673.219001
8   23                 Comedy         831516             4   51960.422475
9   24          Entertainment        3014479             0   28789.341536
10  25        News & Politics         552150             0    7863.682135
11  26          Howto & Style         443219             0   25502.293971
12  27              Education         282663             0   21162.867810
13  28   Science & Technology         521873             0   37068.916017
14  29  Nonprofits & Activism        1919980             0  138413.689189
15  30                 Movies          67232          9336   42150.833333
16  43                  Shows          25196           122    4752.120968

#### 3A. Data Dislikes Video di Canada

In [46]:
# memasukkan data ke dalam list
data_dislikes_ca = {
    
    # a adalah hasil sorting array saat menemukan value yang unique dari id kategori data video region Canada
    'id' : a,
    'highest_dislikes':[gCAvideo.get_group(items)['dislikes'].max() for items in dfCAcat1['id']],
    'lowest_dislikes':[gCAvideo.get_group(items)['dislikes'].min() for items in dfCAcat1['id']],
    'average_dislikes':[gCAvideo.get_group(items)['dislikes'].mean() for items in dfCAcat1['id']
                   ]}

dfDislikesCA = pd.DataFrame(data_dislikes_ca)

In [47]:
dfDislikesCA = pd.merge(dfCAcat1,dfDislikesCA,on="id")

print("")
print("DATA DISLIKES SETIAP KATEGORI DI REGION CANADA")
dfDislikesCA


DATA DISLIKES SETIAP KATEGORI DI REGION CANADA


id                  title  highest_dislikes  lowest_dislikes  \
0    1       Film & Animation             60025                0   
1    2       Autos & Vehicles             12290                1   
2   10                  Music            165854                0   
3   15         Pets & Animals              7732                5   
4   17                 Sports            164693                0   
5   19        Travel & Events              7688                0   
6   20                 Gaming            174645                0   
7   22         People & Blogs            504340                0   
8   23                 Comedy             43698                0   
9   24          Entertainment           1602383                0   
10  25        News & Politics            126649                0   
11  26          Howto & Style             48163                0   
12  27              Education             13561                0   
13  28   Science & Technology             45910                0   
14  29  Nonprofits & Activism            461660                0   
15  30                 Movies              3784              222   
16  43                  Shows              2864               38   

    average_dislikes  
0        1510.319903  
1         560.271955  
2        5922.859823  
3         501.363144  
4        1179.177969  
5         331.278061  
6        1820.596726  
7        2309.572716  
8        1423.937185  
9        1917.447253  
10        784.813898  
11        862.242152  
12        800.653885  
13       1550.781818  
14      31429.824324  
15       2046.666667  
16        971.258065

#### 4A. Data Jumlah Komentar Video di Canada

In [48]:
# memasukkan data ke dalam list
data_comments_ca = {
    
    # a adalah hasil sorting array saat menemukan value yang unique dari id kategori data video region Canada
    'id' : a,
    'highest_comments':[gCAvideo.get_group(items)['comment_count'].max() for items in dfCAcat1['id']],
    'lowest_comments':[gCAvideo.get_group(items)['comment_count'].min() for items in dfCAcat1['id']],
    'average_comments':[gCAvideo.get_group(items)['comment_count'].mean() for items in dfCAcat1['id']
                   ]}

dfCommentsCA = pd.DataFrame(data_comments_ca)

In [49]:
dfCommentsCA = pd.merge(dfCAcat1,dfCommentsCA,on="id")

print("")
print("DATA COMMENTS SETIAP KATEGORI DI REGION CANADA")
dfCommentsCA


DATA COMMENTS SETIAP KATEGORI DI REGION CANADA


id                  title  highest_comments  lowest_comments  \
0    1       Film & Animation            105587                0   
1    2       Autos & Vehicles             29574                0   
2   10                  Music           1114800                0   
3   15         Pets & Animals             45306                0   
4   17                 Sports             63560                0   
5   19        Travel & Events             13612                0   
6   20                 Gaming            122659                0   
7   22         People & Blogs            349112                0   
8   23                 Comedy             83116                0   
9   24          Entertainment            827755                0   
10  25        News & Politics             64291                0   
11  26          Howto & Style            285909                0   
12  27              Education             88591                0   
13  28   Science & Technology            162654                0   
14  29  Nonprofits & Activism            625010                0   
15  30                 Movies              3851              438   
16  43                  Shows              3628               27   

    average_comments  
0        4896.394660  
1        2878.719547  
2       14053.940499  
3        4941.837398  
4        2748.956584  
5        1556.002551  
6        7191.136161  
7        4097.003898  
8        5205.082428  
9        4112.187644  
10       2226.445780  
11       4156.909317  
12       2995.374369  
13       6241.748052  
14      43547.851351  
15       2329.833333  
16        622.927419

## Statistik Data Video Region Great Britain
Di sini kita akan menemukan statistik data video dari region Great Britain menggunakan list <b>gGBvideo</b>.<br/>
Untuk menemukan rata-rata digunakan fungsi <b>.max()</b> pada data grup id kategori yang diinginkan. Untuk menemukan value tertinggi digunakan <b>.max()</b> dan untuk menemukan value terkecil digunakan <b>.min()</b>

Perbedaan data video di Great Britain adalah tidak ada kategori 30 / Movies seperti pada data Canada.

#### 1B. Data Viewer Great Britain

In [50]:
# memasukkan data ke dalam list
data_view_gb = {
    
    # b adalah hasil sorting array saat menemukan value yang unique dari id kategori data video region Great Britain
    'id' : b,
    'highest_views':[gGBvideo.get_group(items)['views'].max() for items in dfGBcat1['id']],
    'lowest_views':[gGBvideo.get_group(items)['views'].min() for items in dfGBcat1['id']],
    'average_views':[gGBvideo.get_group(items)['views'].mean() for items in dfGBcat1['id']
                   ]}

dfViewGB = pd.DataFrame(data_view_gb)

In [51]:
dfViewGB = pd.merge(dfGBcat1,dfViewGB,on="id")

print("")
print("DATA VIEWER SETIAP KATEGORI DI REGION GREAT BRITAIN")
dfViewGB


DATA VIEWER SETIAP KATEGORI DI REGION GREAT BRITAIN


id                  title  highest_views  lowest_views   average_views
0    1       Film & Animation       42560014          2441  3245132.955374
1    2       Autos & Vehicles        5362914          1858  1487392.458333
2   10                  Music      424538912          2152 12444442.690781
3   15         Pets & Animals        6416920         25232  1013708.938202
4   17                 Sports       22975455          1930  1752352.487677
5   19        Travel & Events        6362355          3254  1383177.010417
6   20                 Gaming       18158133          7350   976856.561521
7   22         People & Blogs       62338362          1184  1966605.546480
8   23                 Comedy       43460605         10676  2167430.056346
9   24          Entertainment      169884583          2650  3264607.961530
10  25        News & Politics       18994966           851  1150311.326531
11  26          Howto & Style        8721841          1505   617300.428942
12  27              Education       11240174          8851   972189.402626
13  28   Science & Technology       21530325         38353  3168703.196911
14  29  Nonprofits & Activism       26703269         19270  3919980.688889
15  43                  Shows        1709880        876996  1429164.150000

#### 2B. Data Likes Video di Great Britain

In [52]:
# memasukkan data ke dalam list
data_likes_gb = {
    
    # b adalah hasil sorting array saat menemukan value yang unique dari id kategori data video region Great Britain
    'id' : b,
    'highest_likes':[gGBvideo.get_group(items)['likes'].max() for items in dfGBcat1['id']],
    'lowest_likes':[gGBvideo.get_group(items)['likes'].min() for items in dfGBcat1['id']],
    'average_likes':[gGBvideo.get_group(items)['likes'].mean() for items in dfGBcat1['id']
                   ]}

dfLikesGB = pd.DataFrame(data_likes_gb)

In [53]:
dfLikesGB = pd.merge(dfGBcat1,dfLikesGB,on="id")

print("")
print("DATA LIKES SETIAP KATEGORI DI REGION GREAT BRITAIN")
dfLikesGB


DATA LIKES SETIAP KATEGORI DI REGION GREAT BRITAIN


id                  title  highest_likes  lowest_likes  average_likes
0    1       Film & Animation         461135             0   57205.523089
1    2       Autos & Vehicles         114841            61   24608.506944
2   10                  Music        5613827             0  272138.508943
3   15         Pets & Animals         187524           147   28629.913858
4   17                 Sports        1291139             0   39270.313057
5   19        Travel & Events          23791            54    9674.447917
6   20                 Gaming        1031695             0   39220.127517
7   22         People & Blogs         625174             0   46273.495899
8   23                 Comedy        1253154            22   78431.168490
9   24          Entertainment        3312868             0   81572.362012
10  25        News & Politics         610526            12   25021.685714
11  26          Howto & Style         341145             5   26639.183091
12  27              Education         383049            94   38440.687090
13  28   Science & Technology         392249             0   61531.127413
14  29  Nonprofits & Activism        2061275           708  271695.733333
15  43                  Shows          44588         34615   41836.900000

#### 3B. Data Disikes Video di Great Britain

In [54]:
# memasukkan data ke dalam list
data_dislikes_gb = {
    
    # a adalah hasil sorting array saat menemukan value yang unique dari id kategori data video region Great Britain
    'id' : b,
    'highest_dislikes':[gGBvideo.get_group(items)['dislikes'].max() for items in dfGBcat1['id']],
    'lowest_dislikes':[gGBvideo.get_group(items)['dislikes'].min() for items in dfGBcat1['id']],
    'average_dislikes':[gGBvideo.get_group(items)['dislikes'].mean() for items in dfGBcat1['id']
                   ]}

dfDislikesGB = pd.DataFrame(data_dislikes_gb)

In [55]:
dfDislikesGB = pd.merge(dfGBcat1,dfDislikesGB,on="id")

print("")
print("DATA DISLIKES SETIAP KATEGORI DI REGION GREAT BRITAIN")
dfDislikesGB


DATA DISLIKES SETIAP KATEGORI DI REGION GREAT BRITAIN


id                  title  highest_dislikes  lowest_dislikes  \
0    1       Film & Animation             37132                0   
1    2       Autos & Vehicles             12647                1   
2   10                  Music            421473                0   
3   15         Pets & Animals              3461                1   
4   17                 Sports            193053                0   
5   19        Travel & Events              2434                0   
6   20                 Gaming             17093                0   
7   22         People & Blogs            258504                0   
8   23                 Comedy             38693                2   
9   24          Entertainment           1944971                0   
10  25        News & Politics             37789                0   
11  26          Howto & Style             13141                3   
12  27              Education              8924                4   
13  28   Science & Technology             19405                0   
14  29  Nonprofits & Activism            545595                5   
15  43                  Shows               361              234   

    average_dislikes  
0        2403.156383  
1        1952.708333  
2       11587.191799  
3         685.505618  
4        5199.032512  
5         361.614583  
6        1288.846756  
7        4102.040328  
8        2533.107768  
9        9656.526852  
10       3085.999184  
11        833.606328  
12       1297.927790  
13       3133.299228  
14      60409.488889  
15        319.800000

#### 4B. Data Comments Video di Great Britain

In [56]:
# memasukkan data ke dalam list
data_comments_gb = {
    
    # b adalah hasil sorting array saat menemukan value yang unique dari id kategori data video region Great Britain
    'id' : b,
    'highest_comments':[gGBvideo.get_group(items)['comment_count'].max() for items in dfGBcat1['id']],
    'lowest_comments':[gGBvideo.get_group(items)['comment_count'].min() for items in dfGBcat1['id']],   
    'average_comments':[gGBvideo.get_group(1)['comment_count'].mean() for items in dfGBcat1['id']
                   ]}

dfCommentsGB = pd.DataFrame(data_comments_gb)

In [57]:
dfCommentsGB = pd.merge(dfGBcat1,dfCommentsGB,on="id")

print("")
print("DATA COMMENTS SETIAP KATEGORI DI REGION GREAT BRITAIN")
dfCommentsGB


DATA COMMENTS SETIAP KATEGORI DI REGION GREAT BRITAIN


id                  title  highest_comments  lowest_comments  \
0    1       Film & Animation             50255                0   
1    2       Autos & Vehicles             12083                0   
2   10                  Music           1228655                0   
3   15         Pets & Animals             18532               23   
4   17                 Sports             91663                0   
5   19        Travel & Events              2716                7   
6   20                 Gaming             31803                0   
7   22         People & Blogs            154958                0   
8   23                 Comedy            205692                4   
9   24          Entertainment           1626501                0   
10  25        News & Politics             86221                0   
11  26          Howto & Style            139582                9   
12  27              Education             19451                0   
13  28   Science & Technology             53786                0   
14  29  Nonprofits & Activism            704824               95   
15  43                  Shows              4335             3448   

    average_comments  
0        6049.674428  
1        6049.674428  
2        6049.674428  
3        6049.674428  
4        6049.674428  
5        6049.674428  
6        6049.674428  
7        6049.674428  
8        6049.674428  
9        6049.674428  
10       6049.674428  
11       6049.674428  
12       6049.674428  
13       6049.674428  
14       6049.674428  
15       6049.674428

## Statistik Data Video Region United States
Di sini kita akan menemukan statistik data video dari region United States menggunakan list <b>gUSvideo</b>.<br/>
Untuk menemukan rata-rata digunakan fungsi <b>.max()</b> pada data grup id kategori yang diinginkan. Untuk menemukan value tertinggi digunakan <b>.max()</b> dan untuk menemukan value terkecil digunakan <b>.min()</b>

Perbedaan data video di United States adalah tidak ada kategori 30 / Movies seperti pada data Canada.

#### 1C. Data Viewer United States

In [58]:
# memasukkan data ke dalam list
data_view_us = {
    
    # c adalah hasil sorting array saat menemukan value yang unique dari id kategori data video region United States
    'id' : c,
    'highest_views':[gUSvideo.get_group(items)['views'].max() for items in dfUScat1['id']],
    'lowest_views':[gUSvideo.get_group(items)['views'].min() for items in dfUScat1['id']],
    'average_views':[gUSvideo.get_group(items)['views'].mean() for items in dfUScat1['id']
                   ]}

dfViewUS = pd.DataFrame(data_view_us)

In [59]:
dfViewUS = pd.merge(dfUScat1,dfViewUS,on="id")

print("")
print("DATA VIEWER SETIAP KATEGORI DI REGION UNITED STATES")
dfViewUS


DATA VIEWER SETIAP KATEGORI DI REGION UNITED STATES


id                  title  highest_views  lowest_views  average_views
0    1       Film & Animation       54863912           943 3106250.200853
1    2       Autos & Vehicles       25244097          2860 1355965.408854
2   10                  Music      225211923          1591 6201003.119592
3   15         Pets & Animals        6187457          3393  831143.466304
4   17                 Sports       29090799           658 2025969.030819
5   19        Travel & Events       23932421           789  854619.611940
6   20                 Gaming       16935442          1237 2620830.630355
7   22         People & Blogs       56111957           884 1531835.428660
8   23                 Comedy       29178096          1807 1480308.420017
9   24          Entertainment      149376127           798 2067883.199016
10  25        News & Politics       10277358           549  592587.737837
11  26          Howto & Style       54155921          1107  983730.116739
12  27              Education        7349435           773  712940.815217
13  28   Science & Technology       42799458           983 1452626.745523
14  29  Nonprofits & Activism       24286474          1456 2963884.070175
15  43                  Shows        1445949         36609  903527.333333

#### 2C. Data Likes United States

In [60]:
# memasukkan data ke dalam list
data_likes_us = {
    
    # c adalah hasil sorting array saat menemukan value yang unique dari id kategori data video region United States
    'id' : c,
    'highest_likes':[gUSvideo.get_group(items)['likes'].max() for items in dfUScat1['id']],  
    'lowest_likes':[gUSvideo.get_group(items)['likes'].min() for items in dfUScat1['id']],  
    'average_likes':[gUSvideo.get_group(items)['likes'].mean() for items in dfUScat1['id']
                   ]}

dfLikesUS = pd.DataFrame(data_likes_us)

In [61]:
dfLikesUS = pd.merge(dfUScat1,dfLikesUS,on="id")

print("")
print("DATA LIKES SETIAP KATEGORI DI REGION UNITED STATES")
dfLikesUS


DATA LIKES SETIAP KATEGORI DI REGION UNITED STATES


id                  title  highest_likes  lowest_likes  average_likes
0    1       Film & Animation         922355             0   70787.836247
1    2       Autos & Vehicles          97030             0   11056.395833
2   10                  Music        5613827             0  218918.199011
3   15         Pets & Animals         178243             6   21055.110870
4   17                 Sports        1121217             0   45363.942502
5   19        Travel & Events          42447             3   12030.462687
6   20                 Gaming         444203             2   84502.183599
7   22         People & Blogs        2032463             0   58135.825234
8   23                 Comedy         865679             6   62582.223315
9   24          Entertainment        3093544             0   53243.325070
10  25        News & Politics         163338             0    7298.364696
11  26          Howto & Style         605954             0   39286.076942
12  27              Education         240989             0   29745.031401
13  28   Science & Technology         564804             0   34374.276551
14  29  Nonprofits & Activism        1988746             0  259923.614035
15  43                  Shows          33261           810   18993.666667

#### 3C. Data Dislikes United States

In [62]:
# memasukkan data ke dalam list
data_dislikes_us = {
    
    # c adalah hasil sorting array saat menemukan value yang unique dari id kategori data video region United States
    'id' : c,
    'highest_dislikes':[gUSvideo.get_group(items)['dislikes'].max() for items in dfUScat1['id']],    
    'lowest_dislikes':[gUSvideo.get_group(items)['dislikes'].min() for items in dfUScat1['id']],
    'average_dislikes':[gUSvideo.get_group(items)['dislikes'].mean() for items in dfUScat1['id']
                   ]}

dfDislikesUS = pd.DataFrame(data_dislikes_us)

In [63]:
dfDislikesUS = pd.merge(dfUScat1,dfDislikesUS,on="id")

print("")
print("DATA DILIKES SETIAP KATEGORI DI REGION UNITED STATES")
dfDislikesUS


DATA DILIKES SETIAP KATEGORI DI REGION UNITED STATES


id                  title  highest_dislikes  lowest_dislikes  \
0    1       Film & Animation             38082                0   
1    2       Autos & Vehicles             12290                0   
2   10                  Music            343541                0   
3   15         Pets & Animals              4899                0   
4   17                 Sports            176903                0   
5   19        Travel & Events              9000                0   
6   20                 Gaming            212976                0   
7   22         People & Blogs            228426                0   
8   23                 Comedy             29865                0   
9   24          Entertainment           1674420                0   
10  25        News & Politics            132439                0   
11  26          Howto & Style             66632                0   
12  27              Education             11270                0   
13  28   Science & Technology             47469                0   
14  29  Nonprofits & Activism            497847                0   
15  43                  Shows               616               66   

    average_dislikes  
0        2590.681450  
1         632.838542  
2        7907.757726  
3         573.238043  
4        2361.339006  
5         846.833333  
6       11241.696450  
7        3173.800935  
8        2091.521840  
9        4314.297772  
10       1680.759550  
11       1320.284370  
12        816.408213  
13       1894.378176  
14      58076.859649  
15        429.964912

#### 4C. Data Comments United States

In [64]:
# memasukkan data ke dalam list
data_comments_us = {
    
    # c adalah hasil sorting array saat menemukan value yang unique dari id kategori data video region United States
    'id' : c,
    'highest_comments':[gUSvideo.get_group(items)['comment_count'].max() for items in dfUScat1['id']], 
    'lowest_comments':[gUSvideo.get_group(items)['comment_count'].min() for items in dfUScat1['id']],   
    'average_comments':[gUSvideo.get_group(items)['comment_count'].mean() for items in dfUScat1['id']
                   ]}

dfCommentsUS = pd.DataFrame(data_comments_us)

In [65]:
dfCommentsUS = pd.merge(dfUScat1,dfCommentsUS,on="id")

print("")
print("DATA COMMENTS SETIAP KATEGORI DI REGION UNITED STATES")
dfCommentsUS


DATA COMMENTS SETIAP KATEGORI DI REGION UNITED STATES


id                  title  highest_comments  lowest_comments  \
0    1       Film & Animation            108849                0   
1    2       Autos & Vehicles             10591                0   
2   10                  Music           1228655                0   
3   15         Pets & Animals             44063                0   
4   17                 Sports             83660                0   
5   19        Travel & Events             13631                0   
6   20                 Gaming            144795                0   
7   22         People & Blogs            383762                0   
8   23                 Comedy             81051                0   
9   24          Entertainment           1361580                0   
10  25        News & Politics             38936                0   
11  26          Howto & Style            203884                0   
12  27              Education             28835                0   
13  28   Science & Technology            180913                0   
14  29  Nonprofits & Activism            658130                0   
15  43                  Shows              2181              220   

    average_comments  
0        7627.744136  
1        2042.830729  
2       19359.764524  
3        2892.070652  
4        5148.185373  
5        2267.440299  
6       18042.488372  
7        7719.013084  
8        6521.718831  
9        7383.229426  
10       2428.400885  
11       5583.586589  
12       3286.378019  
13       4993.721783  
14      84364.859649  
15       1668.719298